dsl_parser/tests/test_import_resolver.py

```diff
@@ -0,0 +1,206 @@
+    CustomImportResolver.__module__, CustomImportResolver.__name__)
+
+
+class CustomImportResolverWithoutInit(AbstractImportResolver):
+    def resolve(self, import_url):
+        pass
+custom_no_init_resolver_class_path = "%s:%s" % (
+    CustomImportResolverWithoutInit.__module__,
+    CustomImportResolverWithoutInit.__name__)
+
+
+class FailedToInitializeCustomImportResolver(AbstractImportResolver):
+    def __init__(self):
+        raise MockCustomImportResolverException('mock exception')
+
+    def resolve(self, import_url):
+        pass
+
+failed_custom_resolver_class_path = "%s:%s" % (
+    FailedToInitializeCustomImportResolver.__module__,
+    FailedToInitializeCustomImportResolver.__name__)
+
+
+class CreateImportResolverTests(testtools.TestCase):
+
+    def _test_create_import_resolver(self,
+                                     resolver_configuration=None,
+                                     expected_resolver=None,
+                                     expected_params_name=None,
+                                     err_msg_regex=None):
+        if expected_resolver:
+            resolver = utils.create_import_resolver(resolver_configuration)
+            self.assertEqual(resolver.__class__, expected_resolver.__class__)
+            if expected_params_name:
+                self.assertEqual(
+                    resolver.__getattribute__(expected_params_name),
+                    expected_resolver.__getattribute__(expected_params_name))
+        else:
+            self.assertRaisesRegexp(utils.ResolverInstantiationError,
+                                    err_msg_regex,
+                                    utils.create_import_resolver,
+                                    resolver_configuration)
+
+    def test_no_configuration_specified(self):
+        self._test_create_import_resolver(
+            expected_resolver=DefaultImportResolver(),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_default_class_path_and_params(self):
+        parameters = {
+            DEFAULT_RESLOVER_RULES_KEY: [{'rules1key': 'rules1value'}]
+        }
+        resolver_configuration = {
+            RESOLVER_IMPLEMENTATION_KEY: default_resolver_class_path,
+            RESLOVER_PARAMETERS_KEY: parameters
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=DefaultImportResolver(**parameters),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_default_class_path_no_params(self):
+        resolver_configuration = {
+            RESOLVER_IMPLEMENTATION_KEY: default_resolver_class_path
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=DefaultImportResolver(),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_params_no_class_path(self):
```

Comment:
do we really want this to succeed? it seems like a very strange configuration to do on purpose, so more likely this is a user error, no?